sdk/ai/azure-ai-projects/samples/agents/async_samples/sample_agents_azure_functions_async.py

```diff
@@ -0,0 +1,106 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import asyncio
+
+"""
+FILE: sample_agents_azure_functions_async.py
+ 
+DESCRIPTION:
+    This sample demonstrates how to use azure function agent operations from
+    the Azure Agents service using a asynchronous client.
+ 
+USAGE:
+    python sample_agents_azure_functions_async.py
+ 
+    Before running the sample:
+ 
+    pip install azure-ai-projects azure-identity
+ 
+    Set this environment variables with your own values:
+    PROJECT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+"""
+
+import os
+from azure.ai.projects.aio import AIProjectClient
+from azure.identity.aio import DefaultAzureCredential
+from azure.ai.projects.models import (
+    AzureFunctionStorageQueue,
+    AzureFunctionTool,
+)
+
+project_client = AIProjectClient.from_connection_string(
+    credential=DefaultAzureCredential(exclude_managed_identity_credential=True, exclude_environment_credential=True),
+    conn_str=os.environ["PROJECT_CONNECTION_STRING"],
+)
+
+
+async def main():
+    async with project_client:
+
+        storage_queue_uri = os.environ["STORAGE_QUEUE_URI"]
+        azure_function_tool = AzureFunctionTool(
+            name="foo",
+            description="Get answers from the foo bot.",
+            parameters={
+                "type": "object",
+                "properties": {
+                    "query": {"type": "string", "description": "The question to ask."},
+                    "outputqueueuri": {"type": "string", "description": "The full output queue uri."},
+                },
+            },
+            input_queue=AzureFunctionStorageQueue(
+                queue_name="azure-function-foo-input",
+                storage_queue_uri=storage_queue_uri,
+            ),
+            output_queue=AzureFunctionStorageQueue(
+                queue_name="azure-function-tool-output",
+                storage_queue_uri=storage_queue_uri,
+            ),
+        )
+
+        agent = await project_client.agents.create_agent(
+            model="gpt-4",
+            name="azure-function-agent-foo",
+            instructions=f"You are a helpful support agent. Use the provided function any time the prompt contains the string 'What would foo say?'. When you invoke the function, ALWAYS specify the output queue uri parameter as '{storage_queue_uri}/azure-function-tool-output'. Always responds with \"Foo says\" and then the response from the tool.",
+            headers={"x-ms-enable-preview": "true"},
```

Comment:
We shall check how long the enable preview flag is needed. Please check with Peter Wiese

Reply:
Removed the flag, it is not needed anymore.